corehq/blobs/metadata.py

```diff
@@ -95,10 +95,19 @@ def get(self, **kw):
         :param parent_id: `BlobMeta.parent_id`
         :param type_code: `BlobMeta.type_code`
         :param name: `BlobMeta.name`
+        :param key: `BlobMeta.key`
         :raises: `BlobMeta.DoesNotExist` if the metadata is not found.
         :returns: A `BlobMeta` object.
         """
-        if set(kw) != {"parent_id", "type_code", "name"}:
+        keywords = set(kw)
+        if 'key' in keywords and keywords != {'key', 'parent_id'}:
```

Comment:
I think this entire `if`-block should be replaced with
```suggestion
        if 'key' in keywords:
            if set(keywords) - {'key'}:
                raise TypeError("key is unique; other filters are superfluous")
```

There is a `UNIQUE INDEX` on `key`; it uniquely identifies a single row (or none).

Reply:
But doesnt it need `parent_id` to determine the correct shard?

Reply:
I didn't think through this very thoroughly, and the above suggestion is bad advice. Obviously this will not work in sharded environments, so `parent_id` is also required. Your implementation is correct, ignore my previous comment.

Reply:
Incidentally, there is a problem with the blob db metadata system as implemented in sharded environments: it is possible to have multiple blobs with the same key as long as their parents all resolve to different shards. However, the backend blob storage will still only have one blob per key, and the last one written wins. I have punted on resolving this for now, although maybe it's a serious enough issue it should be addressed sooner rather than later?

Reply:
I have mixed feelings about urgency there. I think from a logical perspective it makes sense that two blobs with the same key would refer to the same thing, so that behavior should not come as a surprise, but if the other situation (same parent_id and key) raises an exception and this one doesn't, that is certainly confusing. Generally speaking, I trust your judgement on how important an issue that is. If it would only happen from a uuid collision, probably safe to punt, but if it is common in the code (even if only old code) to have non-unique parent_ids and generate your own key, then it could be worth fixing. Although, it is possible, and maybe likely, that if you are doing that and your keys collide then you intend to overwrite the data.